anaSbitReadout.py

```diff
@@ -0,0 +1,376 @@
+
+            vfatN[0] = (7 - sbitAddr/192 + ((sbitAddr % 192)/64)*8)
+            sbitSize[0] = ((word >> 11) & 0x7)
+            chHitPerCluster[0] = 2*(sbitSize[0]+1)
+            L1Delay[0] = ((word >> 14) & 0xFFF)
+            eta = vfat_to_etaphi[vfatN[0]][0]
+            phi = vfat_to_etaphi[vfatN[0]][1]
+            # SBIT always includes doublet of adjacent channels
+            vfatCH[0] = 2*(sbitAddr % 64)
+            vfatCH[1] = vfatCH[0] + 1
+            strip[0] = vfat_ch_strips[vfatN[0]]['Strip'][vfatCH[0]]
+            strip[1] = vfat_ch_strips[vfatN[0]]['Strip'][vfatCH[1]]
+            # In case of wrong mapping adjacent channels may not be adjacent strips, which is physically inconsistent
+            if(np.abs(strip[0] - strip[1]) > 1):
+                print 'WARNING: not adjacent strips'
+                time.sleep(3)
+
+            # filling vfat 1Dhistos
+            vfat_h_strip[vfatN[0]].Fill(strip[0])
+            vfat_h_strip[vfatN[0]].Fill(strip[1])
+            vfat_h_ch[vfatN[0]].Fill(vfatCH[0])
+            vfat_h_ch[vfatN[0]].Fill(vfatCH[1])
+            vfat_h_delay[vfatN[0]].Fill(L1Delay[0])
+            vfat_h_sbitSize[vfatN[0]].Fill(sbitSize[0])
+
+            # filling ieta 1Dhistos
+            ieta_h_strip[eta].Fill((phi-1)*128+strip[0])
+            ieta_h_strip[eta].Fill((phi-1)*128+strip[1])
+            ieta_h_ch[eta].Fill((phi-1)*128+vfatCH[0])
+            ieta_h_ch[eta].Fill((phi-1)*128+vfatCH[1])
+            ieta_h_delay[eta].Fill(L1Delay[0])
+            ieta_h_sbitSize[eta].Fill(sbitSize[0])
+
+            # filling 2Dhisto
+            hh_ieta_strip[0].Fill(strip[0]+128*(phi-1), eta)
+            hh_ieta_strip[0].Fill(strip[1]+128*(phi-1), eta)
+            hh_ieta_ch[0].Fill(vfatCH[0]+128*(phi-1), eta)
+            hh_ieta_ch[0].Fill(vfatCH[1]+128*(phi-1), eta)
+
+            """
+            A single sbit word can specify up to 16 adjacent channel hits
+            The following loop takes care of this possibility
+            """
+            for i in range(2, chHitPerCluster[0]):
+                # filling the adjacent channel
+                vfatCH[i] = vfatCH[i-1] + 1
+                # if the new channel exceeds the total VFAT channels, increase phi and move to the first cannel of the next VFAT
+                if vfatCH[i] >= 128 and phi < 3:
+                    phi = phi + 1
+                    vfatCH[i] = 0
+                    vfatN[0] = etaphi_to_vfat[eta][phi]
+                # if the maximum of phi is reached (so there is no "next VFAT"), there must be some kind of error
+                elif vfatCH[i] >= 128 and phi >= 3:
+                    # ERROR
+                    print "ERROR"
+                    print "word", word
+                    print "VFATN:", vfatN[0]
+                    print "VFATCH:", vfatCH[0]
+                    print "Cluster Sz:", sbitSize[0]
+                    print "Eta: ", eta
+                    print "phi: ", phi
+                    break
```

Comment:
This error message is not telling the user why it's an error; moreover how could `vfatCH` and `phi` be respectively greater than 128 and 3?

Suggest to redo this block (`python3` compatible) as a single line with `format(...)` used and a descriptive message to the user.

Reply:
- vfatCH >128 see [here](https://github.com/cms-gem-daq-project/gem-plotting-tools/pull/161#discussion_r234820867)
- phi > 3 it may be due to wrong mapping file or something else